stl/cpp_on_shore/chris/src/task_3/rb_tree.h

```diff
@@ -0,0 +1,353 @@
+#include <assert.h>
+
+#include <iostream>
+
+typedef enum { RED = 0, BLACK } Color;
+
+// 节点类型
+template <typename Type>
+struct RbTreeNode {
+  Color color;         // 颜色
+  Type key;            // 关键字
+  RbTreeNode* left;    // 左子节点
+  RbTreeNode* right;   // 右子节点
+  RbTreeNode* parent;  // 父节点
+};
+
+template <typename Type>
+class RbTree {
+ public:
+  // constructor
+  RbTree() {
+    Nil = BuyNode();
```

Comment:
BuyNode?

Reply:
Now I use a more objective name for this func, called ApplyNode.